Octokit/Models/Request/Enterprise/NewPreReceiveEnvironment.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Diagnostics;
+using System.Globalization;
+using Octokit.Internal;
+
+namespace Octokit
+{
+    /// <summary>
+    /// Describes a new pre-receive environment.
+    /// </summary>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class NewPreReceiveEnvironment
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NewPreReceiveEnvironment"/> class.
+        /// </summary>
+        /// <param name="name">The name of the environment as displayed in the UI.</param>
+        /// <param name="imageUrl">URL to the tarball that will be downloaded and extracted.</param>
+        public NewPreReceiveEnvironment(string name, string imageUrl)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(name, nameof(name));
+            Ensure.ArgumentNotNullOrEmptyString(imageUrl, nameof(imageUrl));
+
+            Name = name;
+            ImageUrl = imageUrl;
+        }
+
+        /// <summary>
+        /// The name of the environment as displayed in the UI.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// URL to the tarball that will be downloaded and extracted.
+        /// </summary>
+        [Parameter(Key = "image_url")]
```

Comment:
FYI, the octokit serializer/deserializer will automatically handle converting from ruby snake case (`image_url`) to c# camel case `ImageUrl` based on the capitalization converting to underscores... so you don't need to specify `[Parameter()]` attributes unless there is a mismatch.  So most of the attributes added in this PR can actually be removed.  (note, the exception is on `Enum` members, where the attribute is still required!)

Reply:
Nice to find a serializer that accounts for that, I automatically assumed I would need an attribute;)